manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -885,6 +885,298 @@ data:
           securityContext:
             fsGroup: 1337
           {{- end }}
+      grpc-simple: |
+        spec:
+          initContainers:
+            - name: grpc-bootstrap-init
```

Comment:
What is the grpc init container for? Is there a design document that describes the detailed workflow and architecture?

Reply:
I'll retroactively throw something together. `grpc-simple` is really only to be used for basic examples. It generates a minimal bootstrap file for [gRPC's XDS support](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md)

Reply:
I browsed the link "gRPC's XDS support" but did not find the architecture and the workflow. Can you describe the the architecture and the detailed workflow, which will help understand how individual components work together to achieve proxyless gRPC? 

In addition, any user guides for configuring, using, and verifying the proxyless gRPC implemented in the PR?

Reply:
User guides will be in a follow-up. Really all the user should have to do is annotate `inject.istio.io/templates=grpc-agent` then they can use `xds:///fqdn` to take advantage of the functionality.

Reply:
The general flow is:

1. Generate a [bootstrap file](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md#xdsclient-and-bootstrap-file); this points to the control plane, describes how control plane auth is handled, and may add "cert providers" which can be used elsewhere in the config (see the mTLS config in listener gen)
2. Reference the bootstrap file using the GRPC_XDS_BOOTSTRAP env on the application container. 
3. Create a gRPC client in the app and make a requests with the scheme `xds:///` this will trigger a discovery request and cache some of the results, using the XDS response for discovery, load-balancing and optionally (m)TLS config

Reply:
The purpose of the `grpc-agent` template is to let istio-agent handle:
* Bootstrap generation so that it includes:
  * node meta
  * cert locations (if OUTPUT_CERTS is enabled)
* SDS/cert generation so gRPC can simply watch files (that's whats supported right now)
* XDS proxy, gRPC doesn't need to support istio-style JWT control-plane auth

Reply:
Regarding to "3. Create a gRPC client in the app", is the "gRPC client" created in this step a sidecar for the app that runs in parallel with the app? Or is the "gRPC client" a gRPC library that compiles with the app?

Reply:
That is via the gRPC library. For Go, they need a side-effect import and the bootstrap env var set: 

https://github.com/grpc/grpc-go/blob/master/examples/features/xds/README.md

Reply:
Are the xds requests from the gPRC library sent to istiod or to istio-proxy, which forwards the requests to istiod?

Reply:
to istio-proxy/istio agent